projects/coordinator/src/test/java/org/batfish/coordinator/ApiKeyAuthenticationFilterTest.java

```diff
@@ -0,0 +1,77 @@
+package org.batfish.coordinator;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Response;
+import org.batfish.common.BatfishLogger;
+import org.batfish.common.CoordConsts;
+import org.batfish.coordinator.config.Settings;
+import org.glassfish.jersey.jackson.JacksonFeature;
+import org.glassfish.jersey.media.multipart.MultiPartFeature;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.glassfish.jersey.test.TestProperties;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/** Tests for {@link ApiKeyAuthenticationFilterTest} */
+public class ApiKeyAuthenticationFilterTest extends JerseyTest {
+
+  @Rule public TemporaryFolder _folder = new TemporaryFolder();
+
+  @Before
+  public void initEnvironment() throws Exception {
+    BatfishLogger logger = new BatfishLogger("debug", false);
+    Settings settings = new Settings(new String[] {});
+    Main.mainInit(new String[] {"-containerslocation", _folder.getRoot().toString()});
+    Main.setLogger(logger);
+    Main.initAuthorizer();
+    Main.setWorkMgr(new WorkMgr(settings, logger));
+  }
+
+  @Override
+  protected Application configure() {
+    forceSet(TestProperties.CONTAINER_PORT, "0");
+    return new ResourceConfig(WorkMgrServiceV2.class)
```

Comment:
for testing a filter don't use the `WorkMgrServiceV2.class` at all. Just make up a resource in this file.